packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/NodeAttributeEditor.tsx

```diff
@@ -43,17 +43,27 @@ export default function NodeAttributeEditor<P extends object>({
 }: NodeAttributeEditorProps<P>) {
   const domApi = useDomApi();
 
+  const initialPropValue = (node as any)[namespace]?.[name];
+  const [propValue, setPropValue] = React.useState<BindableAttrValue<unknown> | null>(
+    initialPropValue,
+  );
+
+  React.useEffect(() => {
+    setPropValue(initialPropValue);
+  }, [initialPropValue]);
+
   const handlePropChange = React.useCallback(
     (newValue: BindableAttrValue<unknown> | null) => {
-      domApi.update((draft) =>
-        appDom.setNodeNamespacedProp(draft, node, namespace as any, name, newValue),
-      );
+      setPropValue(newValue);
+      React.startTransition(() => {
```

Comment:
Nice use of this transition API, we could probably use it more in this type of updates.

Reply:
Ideally we use it just once in the `update` method and eliminate all synchronous coupling between app state and its update method